### PR TITLE
Upgrade Quarto and use notebook metadata for frontmatter 

### PR DESCRIPTION
**Pull request by jackgerrits**
## Why are these changes needed?

- Metadata based front matter is way less error prone and easier to consume
- This should also fix a number of notebook rendering issues that were happening. Thanks Quarto team for the really fast fix! (https://github.com/quarto-dev/quarto-cli/issues/8919)

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by ekzhu — APPROVED** [review]

**Comment by ekzhu** [comment]
Once this is merged can you send an update to the #documentation discord channel to announce the usage? and document-contributor role.

**Comment by jackgerrits** [comment]
Certainly

File: notebook/contributing.md [hunk 1]
```diff
@@ -2,16 +2,25 @@
...
+            "tags": ["code generation", "debugging"],
+            "description": "Use conversable language learning model agents to solve tasks and provide automatic feedback through a comprehensive example of writing, executing, and debugging Python code to compare stock price changes."
+        }
+    }
+}
 ```
 
+**Note**: Notebook metadata can be edited by opening the notebook in a text editor (Or "Open With..." -> "Text Editor" in VSCode)
+
+Ensure the first cell is markdown and before absolutely anything else include the following yaml within a comment.
```
Comment (ekzhu) [inline notebook/contributing.md:22]: Which yaml is it referring to?  I can't find it.

File: notebook/agentchat_groupchat_finite_state_machine.ipynb [hunk 2]
```diff
@@ -597,6 +591,10 @@
   }
  ],
  "metadata": {
+  "front_matter": {
+    "tags": ["orchestration"],
+    "description": "Explore the demonstration of the Finite State Machine implementation, which allows the user to input speaker transition contraints."
```
Comment (qingyun-wu) [inline notebook/agentchat_groupchat_finite_state_machine.ipynb:596]:
```suggestion
    "description": "Explore the demonstration of the Finite State Machine implementation, which allows the user to input speaker transition constraints."
```

File: notebook/agentchat_groupchat_finite_state_machine.ipynb [hunk 3]
```diff
@@ -5,12 +5,6 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "<!--\n",
-    "tags: [\"orchestration\"]\n",
-    "description: |\n",
-    "    Explore the demonstration of the Finite State Machine implementation, which allows the user to input speaker transition contraints.\n",
-    "-->\n",
-    "\n",
     "# FSM - User can input speaker transition contraints.\n",
```
Comment (qingyun-wu) [inline notebook/agentchat_groupchat_finite_state_machine.ipynb:8]:
```suggestion
    "# FSM - User can input speaker transition constraints.\n",
```

**Review by qingyun-wu — COMMENTED** [review]
Fixing some typos. I also have confusion about the referred yaml in the sentence "Ensure the first cell is markdown and before absolutely anything else include the following yaml within a comment." Otherwise, look good!

**Review by qingyun-wu — APPROVED** [review]

**Pull request merged**